the_wizz/kdtree_utils.py

```diff
@@ -204,3 +204,175 @@ def get_mean_array(self):
+    DEC : array_like
+        List of declinations in degrees.
+    leafsize : int
+        The number of points at which the algorithm switches over to
+        brute-force.
+    """
+
+    def __init__(self, RA, DEC, leafsize=16):
+        # convert angular coordinates to 3D points on unit sphere
+        pos_sphere = self._position_sky2sphere(RA, DEC)
+        self.tree = cKDTree(pos_sphere, leafsize)
+
+    @staticmethod
+    def _position_sky2sphere(RA, DEC):
+        """
+        _position_sky2sphere(RA, DEC)
+
+        Maps celestial coordinates onto a unit-sphere in three dimensions
+        (x, y, z).
+
+        Parameters
+        ----------
+        RA : float or array_like
+            Single or list of right ascensions in degrees.
+        DEC : float or array_like
+            Single or list of declinations in degrees.
+
+        Returns
+        -------
+        pos_sphere : array like
+            Data points (x, y, z) representing input points on the unit-sphere,
+            shape of output is (3,) for a single input point or (N, 3) for a
+            set of N input points.
+        """
+        ras_rad = np.deg2rad(RA)
+        decs_rad = np.deg2rad(DEC)
+        try:
+            pos_sphere = np.empty((len(RA), 3))
+        except TypeError:
+            pos_sphere = np.empty((1, 3))
+        cos_decs = np.cos(decs_rad)
+        pos_sphere[:, 0] = np.cos(ras_rad) * cos_decs
+        pos_sphere[:, 1] = np.sin(ras_rad) * cos_decs
+        pos_sphere[:, 2] = np.sin(decs_rad)
+        return np.squeeze(pos_sphere)
+
+    @staticmethod
+    def _distance_sky2sphere(dist_sky):
+        """
+        _distance_sky2sphere(dist_sky)
+
+        Converts angular separation in celestial coordinates to the
+        Euclidean distance in (x, y, z) space.
+
+        Parameters
+        ----------
+        dist_sky : float or array_like
+            Single or list of separations in celestial coordinates.
+
+        Returns
+        -------
+        dist_sphere : float or array_like
+            Celestial separation converted to (x, y, z) Euclidean distance.
+        """
+        dist_sky_rad = np.deg2rad(dist_sky)
+        dist_sphere = np.sqrt(2.0 - 2.0 * np.cos(dist_sky_rad))
+        return dist_sphere
+
+    @staticmethod
+    def _distance_sphere2sky(dist_sphere):
+        """
+        _distance_sphere2sky(dist_sphere)
+
+        Converts Euclidean distance in (x, y, z) space to angular separation in
+        celestial coordinates.
+
+        Parameters
+        ----------
+        dist_sphere : float or array_like
+            Single or list of Euclidean distances in (x, y, z) space.
+
+        Returns
+        -------
+        dist_sky : float or array_like
+            Euclidean distance converted to celestial angular separation.
+        """
+        dist_sky_rad = np.arccos(1.0 - dist_sphere**2 / 2.0)
+        dist_sky = np.rad2deg(dist_sky_rad)
+        return dist_sky
+
+    def query_radius(self, RA, DEC, r):
```

Comment:
For these queries, you'll notice in my function `PairMaker._query_tree` takes in a cosmological distance not a angle as you mean here. Make sure this is clear in the doc string for all these functions. Perhaps we should adopt a convention of r=cosmo distance, ang=angular separation, dist=euclidean distance.